server/controllers/profile.js

```diff
@@ -38,4 +38,29 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
       profile: profile,
     },
   });
-});
\ No newline at end of file
+});
+
+// @route GET /profile/1
+// @desc Get public profile data by id
+// @access Public
+exports.getProfile = asyncHandler(async (req, res, next) => {
+  const searchableId = req.params.id;
+  let profile;
+  try {
+    // Leaving out validating the mongodb ID object for now
+    profile = await Profile.findById({ _id: searchableId }, "name description photo");
```

Comment:
Could we utilize the `express-validation` middleware we have installed already in this PR? Check out the example in the middleware folder to see how it's used.

Reply:
Thanks! I added the validate-record-id logic to the middleware layer